gix-path/src/env/mod.rs

```diff
@@ -102,6 +106,36 @@ pub fn xdg_config(file: &str, env_var: &mut dyn FnMut(&str) -> Option<OsString>)
         })
 }
 
+static GIT_CORE_DIR: Lazy<Option<PathBuf>> = Lazy::new(|| {
+    let mut cmd = std::process::Command::new(exe_invocation());
+
+    #[cfg(windows)]
+    {
+        use std::os::windows::process::CommandExt;
+        const CREATE_NO_WINDOW: u32 = 0x08000000;
+        cmd.creation_flags(CREATE_NO_WINDOW);
+    }
+    let output = cmd.arg("--exec-path").output().ok()?;
+
+    if !output.status.success() {
+        return None;
+    }
+
+    BString::new(output.stdout)
+        .trim_with(|b| b.is_ascii_whitespace())
```

Comment:
I this this change is sufficient to fix #1760:

```suggestion
        .strip_suffix(b"\n")?
```

As noted there, there may be a benefit to fixing that here, but it is definitely fine to leave if for later. I've put more details in the issue, including the (limited) testing I've done of this change.

Reply:
I will apply it - the `is_ascii_whitespace` I copied from somewhere and knew it would replace the version that is now re-added with the patch. It's good to know that this is the more correct version.